lib/class-wp-theme-json-gutenberg.php

```diff
@@ -141,77 +141,72 @@ class WP_Theme_JSON_Gutenberg {
-	 *
-	 * - css_var_infix => infix to use in generating the CSS Custom Property. Example:
-	 *                   --wp--preset--<preset_infix>--<slug>: <preset_value>
-	 *
-	 * - classes      => array containing a structure with the classes to
-	 *                   generate for the presets. Each class should have
-	 *                   the class suffix and the property name. Example:
-	 *
-	 *                   .has-<slug>-<class_suffix> {
-	 *                       <property_name>: <preset_value>
-	 *                   }
+	 * - path       => where to find the preset within the settings section
+	 *
+	 * - value_key  => the key that represents the value
+	 *
+	 * - value_func => optionally, instead of value_key, a function to generate
+	 *                 the value that takes a preset as an argument
+	 *
+	 * - css_var    => name of the var to generate. The "$slug" substring will be
+	 *                 replaced by the slug of each preset. For example,
+	 *                 given a preset for color with two values whose slugs are "black" and "white",
+	 *                 the string "--wp--preset--color--$slug" will generate two variables:
+	 *                 "--wp--preset--color--black" and "--wp--preset--color--white".
+	 *
+	 * - classes    => array containing a structure with the classes to
+	 *                 generate for the presets, where for each array item
+	 *                 the key is the class name and the value the property name.
+	 *                 The "$slug" substring will be replaced by the slug of each preset.
+	 *                 For example:
+	 *                 'classes' => array(
+	 *                   '.has-$slug-color'            => 'color',
+	 *                   '.has-$slug-background-color' => 'background-color',
+	 *                   '.has-$slug-border-color'     => 'border-color',
+	 *                 )
+	 * - properties => array of CSS properties to be used by kses to
+	 *                 validate the content of each preset
+	 *                 by means of the remove_insecure_properties method.
 	 */
 	const PRESETS_METADATA = array(
 		array(
-			'path'          => array( 'color', 'palette' ),
-			'value_key'     => 'color',
-			'css_var_infix' => 'color',
-			'classes'       => array(
-				array(
-					'class_suffix'  => 'color',
-					'property_name' => 'color',
-				),
-				array(
-					'class_suffix'  => 'background-color',
-					'property_name' => 'background-color',
-				),
-				array(
-					'class_suffix'  => 'border-color',
-					'property_name' => 'border-color',
-				),
+			'path'       => array( 'color', 'palette' ),
+			'value_key'  => 'color',
+			'css_vars'   => '--wp--preset--color--$slug',
+			'classes'    => array(
+				'.has-$slug-color'            => 'color',
+				'.has-$slug-background-color' => 'background-color',
+				'.has-$slug-border-color'     => 'border-color',
 			),
+			'properties' => array( 'color', 'background-color', 'border-color' ),
```

Comment:
Should we introduce the usage of "properties" in this PR as well? It seems weird to add the properties but not use them? It's not clear what they're here for.

Reply:
They're used [here](https://github.com/WordPress/gutenberg/pull/35228/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7R1312), within the `remove_insecure_settings` method.

Reply:
Oh I missed it sorry.